test/unit/org/apache/cassandra/metrics/CompactionMetricsTest.java

```diff
@@ -0,0 +1,229 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.compaction.ActiveCompactionsTracker;
+import org.apache.cassandra.db.compaction.CompactionController;
+import org.apache.cassandra.db.compaction.CompactionInfo;
+import org.apache.cassandra.db.compaction.CompactionIterator;
+import org.apache.cassandra.db.compaction.CompactionManager;
+import org.apache.cassandra.db.compaction.OperationType;
+import org.apache.cassandra.db.lifecycle.LifecycleTransaction;
+import org.apache.cassandra.io.sstable.ISSTableScanner;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.cassandra.schema.MockSchema;
+import org.apache.cassandra.utils.FBUtilities;
+
+import static org.junit.Assert.assertNotNull;
+
+public class CompactionMetricsTest extends CQLTester
+{
+    @Test
+    public void testSimpleCompactionMetricsForCompletedTasks() throws Throwable
+    {
+
+        final long initialCompletedTasks = CompactionManager.instance.getCompletedTasks();
+        Assert.assertEquals(0L, CompactionManager.instance.getTotalCompactionsCompleted());
+        Assert.assertEquals(0L, CompactionManager.instance.getTotalBytesCompacted());
+
+        this.createTable("CREATE TABLE %s (pk int, ck int, a int, b int, PRIMARY KEY (pk, ck))");
+        this.getCurrentColumnFamilyStore().disableAutoCompaction();
+        for (int i = 0; i < 10; ++i)
+        {
+            this.execute("INSERT INTO %s (pk, ck, a, b) VALUES (" + i + ", 2, 3, 4)");
+        }
+
+        this.getCurrentColumnFamilyStore().forceBlockingFlush();
+        this.getCurrentColumnFamilyStore().forceMajorCompaction();
+
+        Assert.assertEquals(1L, CompactionManager.instance.getTotalCompactionsCompleted());
+        Assert.assertTrue(CompactionManager.instance.getTotalBytesCompacted() > 0L);
+        Assert.assertTrue(initialCompletedTasks < CompactionManager.instance.getCompletedTasks());
+    }
+
+    @Test
+    public void testCompactionMetricsForPendingTasks()
+    {
+        final ColumnFamilyStore cfs = MockSchema.newCFS();
+        final List<SSTableReader> sstables = this.createSSTables(cfs, 10, 0);
+        final Set<SSTableReader> toMarkCompacting = new HashSet<>(sstables.subList(0, 3));
+        final TestCompactionTask tct = new TestCompactionTask(cfs, toMarkCompacting);
+        final long initialPendingTasks = CompactionManager.instance.getMetrics().pendingTasks.getValue();
+        try
+        {
+            tct.start();
+            final List<CompactionInfo.Holder> activeCompactions = this.getActiveCompactionsForTable(cfs);
+            Assert.assertEquals(1L, activeCompactions.size());
+            Assert.assertEquals(activeCompactions.get(0).getCompactionInfo().getSSTables(), toMarkCompacting);
+            cfs.runWithCompactionsDisabled(() -> null, sstable -> !toMarkCompacting.contains(sstable), false, false, true);
```

Comment:
Same question as on line 121

Reply:
these were some additional checks to make sure the compactions were started but left in a pending state and not considered failed. will remove them as you pointed it out as being redundant.